plugins/out_stackdriver/stackdriver_conf.c

```diff
@@ -281,6 +298,38 @@ struct flb_stackdriver *flb_stackdriver_conf_create(struct flb_output_instance *
         ctx->resource = flb_sds_create(FLB_SDS_RESOURCE_TYPE);
     }
 
+    flb_kv_init(&ctx->labels);
+
+    /*
+     * Load monitored resource labels from config files.
+     * Will only use labels specified for a particular resource type at
+     * https://cloud.google.com/logging/docs/api/v2/resource-list
+    */
+    mk_list_foreach(labels, &ins->properties) {
+        kv = mk_list_entry(labels, struct flb_kv, _head);
+        if (strncasecmp(kv->key, "label.", 6) == 0 &&
+            flb_sds_len(kv->key) > 6) {
+                /* Get label name from config file label identifier*/
+                tmp = strtok(kv->key, ".");
+                /* Only needs do be done twice to get to the first '.' */
+                tmp = strtok(NULL, ".");
+                if (tmp == NULL || strncasecmp(tmp, "label", 5) == 0) {
```

Comment:
Nope. The way strtok works is that if you have label.something the first time you call it will return "label" and the second time you call it returns "something"

Reply:
Yes, I'm familiar with how `strtok` works (and you should be using `strtok_r` for thread safety, FWIW). By this time you've called it twice (line 313 and line 315), assigning to `tmp` both times, so `tmp` will be the content after the first `'.'` (and before the second, if any).

If you meant to save the result of the first call in another variable, there's no need to compare that to `"label"`, because you're guaranteed (by the test in line 310) that it will be exactly that. But see my other comment about not needing `strtok` at all.

Reply:
removed strtok